### DISCUSSION TITLE
Question: Why I can't see the implements of 'vips_$xx_get_type()' , eg 'vips_system_get_type()' ?

### DISCUSSION THREAD

**Discussion by LightSun** [libvips/libvips#2110] [Q&A]

**Comment by jcupitt** [comment]
Hi @LightSun, the `_get_type()` functions are generated by the `G_DEFINE_TYPE` macros. Have a look at the gobject docs:

https://developer.gnome.org/gobject/stable/gobject-Type-Information.html#G-DEFINE-TYPE:CAPS

**Reply by LightSun** [reply]
thanks
